packages/core/src/NavigationFocusedContext.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { NavigationHelpers, ParamListBase } from './types';
+
+export type FocusedNavigationCallback = (
+  navigation: NavigationHelpers<ParamListBase>
+) => any;
+
+export type FocusedNavigationListener = (
+  callback: FocusedNavigationCallback
+) => { handled: boolean; result: any };
+
+const MISSING_CONTEXT_ERROR =
+  "We couldn't find a navigation focused context. Have you wrapped your app with 'NavigationContainer'?";
+
+/**
+ * Context which holds the required helpers needed to access focused navigator.
+ */
+
+const NavigationFocusedContext = React.createContext<{
```

Comment:
I've removed the need for a `removeXListener`, so maybe we can use the same builder context for this now?

Reply:
can you clarify?

Reply:
Oh you mean to construct useEffect directly here. It makes sense, but I don't have opinion if it's obviously better

Reply:
thanks!